packages/app/src/runs/modals/SelectCloudProjectModal.vue

```diff
@@ -294,7 +294,17 @@ const pickedOrganization = ref(organizations.value.length >= 1 ? organizations.v
 
 const projects = computed(() => pickedOrganization.value?.projects?.nodes || [])
 const newProject = ref(projects.value.length === 0)
-const pickedProject = ref(pickedOrganization.value?.projects ? pickedOrganization.value.projects.nodes.find((p) => p.name === projectName.value) : undefined)
+const pickedProject = computed(() => {
+  if (pickedOrganization.value?.projects) {
+    if (pickedOrganization.value.projects.nodes.length === 1) {
+      return pickedOrganization.value.projects.nodes[0]
+    }
+
+    return pickedOrganization.value.projects.nodes.find((p) => p.name === projectName.value)
```

Comment:
I don't think we can go with id here. In these scenarios, we are trying to tie a local project that doesn't have a `projectId` to a cloud project.

Reply:
@rockindahizzy The naming is a bit deceptive here - a project can have both an `id` and a `projectId`. `id` is a UUID-esque value that uniquely identifies any project, `projectId` is a cloud-assigned slug for connected projects

Reply:
Just for my knowledge, does an id persist forever on a single machine?

Reply:
We cannot use name, it's possible to have two projects with the same name.

![image](https://user-images.githubusercontent.com/19196536/184293126-6f824acc-fb9f-4236-8010-d27d12c75d21.png)

Reply:
Actually this logic makes no sense - where did this even come from? The requirement is 

> In the "connect project" modal in the runs tab, pre-select the existing project if and only if there is only one.

So basically:

- If there is 1 project, select it
- Otherwise, default to "Pick a project" placeholder

Reply:
See: https://github.com/cypress-io/cypress/pull/23266#issuecomment-1212786917

Reply:
It seems like the idea is to narrow down a probable project to reduce user clicks until there are no options. The existing logic was to pre-select a cloud project if it matches the project name, otherwise let the user choose. 
```js
const pickedProject = ref(pickedOrganization.value?.projects ? pickedOrganization.value.projects.nodes.find((p) => p.name === projectName.value) : undefined)
```

I'm assuming if the requirement was to replace the existing logic with this new logic, it would have been called out?

Reply:
Going to keep the name matching logic based on comments [here](https://github.com/cypress-io/cypress/issues/22936#issuecomment-1213223632)